pkg/operator/status/status_controller.go

```diff
@@ -110,12 +110,15 @@ func Add(mgr, rootCredentialManager manager.Manager, kubeConfig string) error {
 	}
 
 	// always reconcile status when the clusteroperator/cloud-credential changes.
-	if err := c.Watch(source.Kind(operatorCache, &configv1.ClusterOperator{}), &handler.EnqueueRequestForObject{}, predicate.Funcs{
-		GenericFunc: func(genericEvent event.GenericEvent) bool {
-			if genericEvent.Object == nil {
-				return false
-			}
-			return genericEvent.Object.GetName() == "cloud-credential"
+	if err := c.Watch(source.Kind(operatorCache, &configv1.ClusterOperator{}), handler.EnqueueRequestsFromMapFunc(alwaysReconcileCCOConfigObject), predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Object.GetName() == constants.CloudCredClusterOperatorName
```

Comment:
I'm not super familiar with how these callbacks work, but is there any chance of a nil dereference here? Maybe add the nil check back to all 3 of the functions, just to be safe

Reply:
Thanks for pointing this out. After a some digging, it looks like `e.Object` or `e.ObjectNew` come from `watch.Event` objects which in turn are sourced from a client-go-style `.Watch().ResultChan()`. These objects are essentially part of the API server's response and should not be nil without the decoder erroring out. 

Nevertheless, it is definitely good to be cautious. nil checks are added in the callbacks for safety.